In [116]:
init_data = [10, 11, 13, 17, 23, 31, 40, 45, 38, 33, 30, 29]

# Прирост базисный.
## $ \Delta_{iб} = y_i - y_1$


In [56]:
list(y - init_data[0] for y in init_data[1:])

[1, 3, 7, 13, 27, 30, 35, 28, 23, 20, 19]

# Цепной прирост.
## $ \Delta_{iц} = y_i - y_{i-1}$

In [46]:
print('-', end=' ')
for i in range(1, len(init_data)):
    print(init_data[i] - init_data[i-1], end=' ')

- 1 2 4 6 14 3 5 -7 -5 -3 -1 

# Коэффицент роста базисный.
## $К_{рб} = \frac{yi}{y}$


In [19]:
print('-', *(round(init_data[i] / init_data[i-1], 1) for i in range(1, len(init_data))))

- 1.1 1.2 1.3 1.4 1.3 1.3 1.1 0.8 0.9 0.9 1.0


# Темп роста базисный.
## $ Т_{рб} = \frac{y_i}{y} \cdot 100\% $

In [35]:
print('-', *(int(round(init_data[i] / init_data[i-1] * 100, 0)) for i in range(1, len(init_data))))

- 110 118 131 135 135 129 112 84 87 91 97


# Коэфициент прироста базисный.
## $К_{пб_i} = \frac{\Delta_{iб}}{y1} = К_{рб} - 1$

In [42]:
print('-', *(round(init_data[i] / init_data[0] - 1, 1) for i in range(0, len(init_data))))

- 0.0 0.1 0.3 0.7 1.3 2.1 3.0 3.5 2.8 2.3 2.0 1.9


# Темп прироста базисный.
## $Т_{пб_i} = К_{пб_i}\cdot100\%$

In [27]:
print('-', *(int(round((init_data[i] / init_data[0] - 1)*100, 1)) for i in range(1, len(init_data))))

- 10 30 70 130 210 300 350 280 230 200 190


# Коэфициент роста цепной.
## $К_{рц_i} = \frac{y_i}{y_{i-1}}$

In [48]:
print('-', *(round(init_data[i] / init_data[i-1], 2) for i in range(1, len(init_data))))

- 1.1 1.18 1.31 1.35 1.35 1.29 1.12 0.84 0.87 0.91 0.97


# Коэфициент прироста цепной.
## $К_{пц_i} = К_{рц_i} - 1$

In [51]:
print('-', *(round((init_data[i] / init_data[i-1]) - 1, 2) for i in range(1, len(init_data))))

- 0.1 0.18 0.31 0.35 0.35 0.29 0.12 -0.16 -0.13 -0.09 -0.03


# Темп роста цепной.
## $Т_{рц_i} = К_{рц_i} \cdot 100\%$

In [53]:
print('-', *(round(init_data[i] / init_data[i-1] *100, 0) for i in range(1, len(init_data))))

- 110.0 118.0 131.0 135.0 135.0 129.0 112.0 84.0 87.0 91.0 97.0


# Темп прироста цепной.
## $Т_{пц_i} = К_{пц_i}\cdot100\%$

In [55]:
print('-', *(round(((init_data[i] / init_data[i-1]) - 1) * 100, 0) for i in range(1, len(init_data))))

- 10.0 18.0 31.0 35.0 35.0 29.0 12.0 -16.0 -13.0 -9.0 -3.0


# Линейная регрессия.
По фактическим значениям строится теоретическая зависимость:

# $\widetilde{y_i} = k \cdot x_i +b = 2.4476 \cdot x_i + 10.757 $

## $k = \frac{\overline{xy} - \overline{x} \cdot \overline{y}}{\overline{x^2}-(\overline{x})^2} = \frac{202,5 - 6,5\cdot26,6(6)}{54,1666-6,5^2} = \frac{29.167}{11.916(6)} = 2.4476$ 
k - среднее произведение минус произведение средних делить на средний квадрат x минус квадрат среднего x
## $b = \overline{y} - k\cdot\overline{x} = 26,6(6) - 2.4476 \cdot 6,5 = 10.757 $

____________________

$\overline{x} = \frac{x_i + x_{i+1} + ... + x_n}{n} = 6,5$ $ 

Среднее арифметическое или первый и последний член последовательности деленый на 2.

Можно применять расчет по первому и последнему члену последовательности если ряд монотонно возрастающий

$\overline{x^2} = \frac{x_i^2 + x_{i+1}^2 + ... + x_n^2}{n} = 54,1666$

$\overline{y} = \frac{y_i + y_{i+1} + ... + y_n}{n} = \frac{320}{12} = 26,6(6)$

$\overline{y^2} = \frac{y_i^2 + y_{i+1}^2 + ... + y_n^2}{n} = 839 $

$\overline{xy} = \frac{xy_i + xy_{i+1} + ... + xy_n}{n} = \frac{2430}{12} = 202,5$


In [117]:
x_avg = 6.5
x2_avg = 54.1666666
y_avg = 320/12
y2_avg = 839
xy_avg = 202.5

k = 2.4476
b = 10.757

# Считаем линейную регрессию с параметрами:

Для каждого элемента x формула будет следующей: $\widetilde{y_i} = k \cdot x_i +b = 2.4476 \cdot x_i + 10.757 $

In [109]:
for x in range(1,12+3):
    print(x, end=' ')
print()
for x in range(1,12+3):
    print(round(k*x+b,1), end=' ')

1 2 3 4 5 6 7 8 9 10 11 12 13 14 
13.2 15.7 18.1 20.5 23.0 25.4 27.9 30.3 32.8 35.2 37.7 40.1 42.6 45.0 

$ \widetilde{y_i} = 13.2\space15.7\space18.1\space20.5\space23.0\space25.4\space27.9\space30.3\space32.8\space35.2\space37.7\space40.1\space42.6\space45.0  $

# Линейный коэфициент корреляции
## $ r_{xy} = \frac{\sum(x_i - \overline{x})\times(y_i - \overline{y})}{\sqrt{\sum(x_i - \overline{x})^2\times\sum(y_i - \overline{y})^2}}$